Xamarin.Forms.Core/ResourceDictionary.cs

```diff
@@ -34,6 +54,39 @@ public class ResourceDictionary : IResourceDictionary, IDictionary<string, objec
 		}
 
 		ResourceDictionary _mergedInstance;
+		ICollection<ResourceDictionary> _mergedDictionaries;
+		public ICollection<ResourceDictionary> MergedDictionaries { get { return _mergedDictionaries; } }
+
+		public ResourceDictionary()
+		{
+			var collection = new ObservableCollection<ResourceDictionary>();
+			collection.CollectionChanged += MergedDictionaries_CollectionChanged;
+			_mergedDictionaries = collection;
+		}
+
+		private void MergedDictionaries_CollectionChanged(object sender, System.Collections.Specialized.NotifyCollectionChangedEventArgs e)
+		{
+			// New Items
+			var newItems = e.NewItems?.Cast<ResourceDictionary>();
+			if (newItems != null)
+			{
+				foreach (var item in newItems)
+					item.ValuesChanged += Item_ValuesChanged;
+
+				if (newItems.Count() > 0)
+					OnValuesChanged(newItems.SelectMany(x => x).ToArray());
+			}
+			// Old Items
+			var oldItems = e.OldItems?.Cast<ResourceDictionary>();
+			if (oldItems != null)
+				foreach (var item in oldItems)
+					item.ValuesChanged -= Item_ValuesChanged;
```

Comment:
Decided to use a collection tracker, to handle the Clear command. Seemed like a code smell to introduce a new List type just to accommodate this. Keeping a secondary list of the collection, should not result in any large memory issues, since its just tracking references, not cloning the objects.